Rebaseline two tracking prevention tests

sysrqb opened the pull request:
#### 3879b0dcdb487af27e6daa50d22e975ef5f5f98b
<pre>
Rebaseline two tracking prevention tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=261989">https://bugs.webkit.org/show_bug.cgi?id=261989</a>
rdar://115921610

Reviewed by John Wilander.

In 251397@main, we aligned tracking prevention&apos;s client side cookie cap with
other script-writable storage. Now let&apos;s update the macOS test expectations and
re-enabled the tests.

* LayoutTests/http/tests/resourceLoadStatistics/website-data-removal-for-site-navigated-to-with-link-decoration-expected.txt:
* LayoutTests/http/tests/resourceLoadStatistics/website-data-removal-for-site-navigated-to-with-link-decoration.html:
* LayoutTests/http/tests/resourceLoadStatistics/website-data-removal-for-site-without-user-interaction-expected.txt:
* LayoutTests/http/tests/resourceLoadStatistics/website-data-removal-for-site-without-user-interaction.html:
* LayoutTests/platform/glib/http/tests/resourceLoadStatistics/website-data-removal-for-site-navigated-to-with-link-decoration-expected.txt: Added.
* LayoutTests/platform/glib/http/tests/resourceLoadStatistics/website-data-removal-for-site-without-user-interaction-expected.txt: Added.
* LayoutTests/platform/mac-monterey/http/tests/resourceLoadStatistics/website-data-removal-for-site-navigated-to-with-link-decoration-expected.txt: Copied from LayoutTests/http/tests/resourceLoadStatistics/website-data-removal-for-site-navigated-to-with-link-decoration-expected.txt.
* LayoutTests/platform/mac-monterey/http/tests/resourceLoadStatistics/website-data-removal-for-site-without-user-interaction-expected.txt: Copied from LayoutTests/http/tests/resourceLoadStatistics/website-data-removal-for-site-without-user-interaction-expected.txt.
* LayoutTests/platform/mac-wk2/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/268681@main">https://commits.webkit.org/268681@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/48f73cf96570a9f557c0f2452cabb5b9959c537e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/20293 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/20725 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/21366 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/22193 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/18937 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/23982 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/20896 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/20369 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/20513 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20413 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/17649 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/23043 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/17582 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/18452 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/24720 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18658 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/18628 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22683 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/19209 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/16309 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/18414 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/18288 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4892 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/22755 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/19034 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

johnwilander commented on LayoutTests/http/tests/resourceLoadStatistics/website-data-removal-for-site-navigated-to-with-link-decoration-expected.txt:
```diff
@@ -7,7 +7,6 @@ Before deletion: LocalStorage entry does exist.
 Before deletion: IDB entry does exist.
 
 After deletion: HttpOnly cookie exists.
-After deletion: Client-side cookie exists.
```
This should instead say "Client-side cookie does not exist." We prefer explicit test output rather than the absence of output meaning something important.

johnwilander commented on LayoutTests/http/tests/resourceLoadStatistics/website-data-removal-for-site-navigated-to-with-link-decoration.html:
```diff
@@ -5,7 +5,7 @@
     <script src="resources/util.js"></script>
 </head>
 <body onload="setTimeout('runTest()', 0)">
-<div id="description">Check that non-cookie website data gets removed after a navigation with link decoration from a prevalent resource.</div>
```
This should be changed to "Check that script-writeable website data …" since we *are* keeping HttpOnly and regular server-set cookies.

johnwilander commented on LayoutTests/http/tests/resourceLoadStatistics/website-data-removal-for-site-without-user-interaction-expected.txt:
```diff
@@ -1,4 +1,4 @@
-Check that non-cookie website data gets removed after a period of no user interaction.
```
Ditto.

johnwilander commented on LayoutTests/http/tests/resourceLoadStatistics/website-data-removal-for-site-without-user-interaction-expected.txt:
```diff
@@ -7,7 +7,6 @@ Before deletion: LocalStorage entry does exist.
 Before deletion: IDB entry does exist.
 
 After deletion: HttpOnly cookie exists.
-After deletion: Client-side cookie exists.
```
Ditto on "does not exist."

johnwilander commented on LayoutTests/http/tests/resourceLoadStatistics/website-data-removal-for-site-without-user-interaction.html:
```diff
@@ -5,7 +5,7 @@
 </head>
 <body onload="setTimeout('runTest()', 0)">
-<div id="description">Check that non-cookie website data gets removed after a period of no user interaction.</div>
+<div id="description">Check that website data gets removed after a period of no user interaction.</div>
```
Ditto.

johnwilander commented on LayoutTests/platform/glib/http/tests/resourceLoadStatistics/website-data-removal-for-site-navigated-to-with-link-decoration-expected.txt:
```diff
@@ -0,0 +1,35 @@
+Check that website data gets removed after a navigation with link decoration from a prevalent resource.
```
Ditto on "script-writeable" here too.

johnwilander commented on LayoutTests/platform/glib/http/tests/resourceLoadStatistics/website-data-removal-for-site-navigated-to-with-link-decoration-expected.txt:
```diff
@@ -0,0 +1,35 @@
+Before deletion: IDB entry does exist.
+
+After deletion: HttpOnly cookie exists.
+After deletion: Client-side cookie exists.
```
Is this expected?

sysrqb replied:
Yes, we only delete script-written cookies on Ventura+, so this is expected on the glib port. I considered marking this test as failing on Monterey/glib, but having test coverage on EWS seems valuable even though it is missing some of the desired behavior. I'll let the Linux folks know about this change.

johnwilander replied:
Got it. Thanks!

johnwilander commented on LayoutTests/platform/glib/http/tests/resourceLoadStatistics/website-data-removal-for-site-without-user-interaction-expected.txt:
```diff
@@ -0,0 +1,15 @@
+Check that website data gets removed after a period of no user interaction.
```
Ditto.

johnwilander commented on LayoutTests/platform/glib/http/tests/resourceLoadStatistics/website-data-removal-for-site-without-user-interaction-expected.txt:
```diff
@@ -0,0 +1,15 @@
+Before deletion: IDB entry does exist.
+
+After deletion: HttpOnly cookie exists.
+After deletion: Client-side cookie exists.
```
Is this expected?

johnwilander replied:
Same thing here?

sysrqb replied:
Yes, sorry that wasn't clear, this is a glib expectation and the old behavior is expected.

johnwilander commented on LayoutTests/platform/mac-monterey/http/tests/resourceLoadStatistics/website-data-removal-for-site-navigated-to-with-link-decoration-expected.txt:
```diff
@@ -0,0 +1,35 @@
+Check that website data gets removed after a navigation with link decoration from a prevalent resource.
```
Ditto.

johnwilander commented on LayoutTests/platform/mac-monterey/http/tests/resourceLoadStatistics/website-data-removal-for-site-navigated-to-with-link-decoration-expected.txt:
```diff
@@ -0,0 +1,35 @@
+Before deletion: IDB entry does exist.
+
+After deletion: HttpOnly cookie exists.
+After deletion: Client-side cookie exists.
```
Is this expected?

sysrqb replied:
Yes, we only delete script-written cookies on Ventura+, so this is expected on Monterey. I considered marking this test as failing on Monterey (and glib), but having test coverage on EWS seems valuable even though it is missing some of the desired behavior.

johnwilander commented on LayoutTests/platform/mac-monterey/http/tests/resourceLoadStatistics/website-data-removal-for-site-without-user-interaction-expected.txt:
```diff
@@ -0,0 +1,15 @@
+Check that website data gets removed after a period of no user interaction.
```
Ditto.

johnwilander commented on LayoutTests/platform/mac-monterey/http/tests/resourceLoadStatistics/website-data-removal-for-site-without-user-interaction-expected.txt:
```diff
@@ -0,0 +1,15 @@
+Before deletion: IDB entry does exist.
+
+After deletion: HttpOnly cookie exists.
+After deletion: Client-side cookie exists.
```
Is this expected?

johnwilander replied:
Same thing here?

sysrqb replied:
Yes, sorry that wasn't clear, this is a mac-monterey expectation and the old behavior is expected.

johnwilander commented on LayoutTests/platform/mac-wk2/TestExpectations:
```diff
@@ -1697,8 +1697,6 @@ webkit.org/b/255407 [ Monterey x86_64 ] imported/w3c/web-platform-tests/html/sem
 [ BigSur Monterey ] fast/images/animated-heics-draw.html [ Skip ]
 [ Ventura+ ] fast/images/animated-heics-draw.html [ Pass Timeout ]
-[ Ventura+ ] http/tests/resourceLoadStatistics/website-data-removal-for-site-without-user-interaction.html [ Failure ]
-[ Ventura+ ] http/tests/resourceLoadStatistics/website-data-removal-for-site-navigated-to-with-link-decoration.html [ Failure ]
```
Nice to get these addressed!

johnwilander commented on LayoutTests/http/tests/resourceLoadStatistics/website-data-removal-for-site-navigated-to-with-link-decoration.html:
```diff
@@ -5,17 +5,17 @@
-    const clientSideCookieName = "client-side-cookie";
+    const httpOnlyCookieName = "http_only_cookie";
+    const serverSideCookieName = "server_side_cookie";
+    const clientSideCookieName = "client_side_cookie";
```
Is this some whitespace or linefeed change?

sysrqb replied:
The change here is replacing "-" with "_" in the string, but as I was writing an explanation about why this was needed I realized it's not needed.

johnwilander reviewed (approved):
See my remaining comments. Assuming we're good there, this looks ready to land.

sysrqb committed:
Rebaseline two tracking prevention tests
https://bugs.webkit.org/show_bug.cgi?id=261989
rdar://115921610

Reviewed by John Wilander.

In 251397@main, we aligned tracking prevention's client side cookie cap with
other script-writable storage. Now let's update the macOS test expectations and
re-enabled the tests.

* LayoutTests/http/tests/resourceLoadStatistics/website-data-removal-for-site-navigated-to-with-link-decoration-expected.txt:
* LayoutTests/http/tests/resourceLoadStatistics/website-data-removal-for-site-navigated-to-with-link-decoration.html:
* LayoutTests/http/tests/resourceLoadStatistics/website-data-removal-for-site-without-user-interaction-expected.txt:
* LayoutTests/http/tests/resourceLoadStatistics/website-data-removal-for-site-without-user-interaction.html:
* LayoutTests/platform/glib/http/tests/resourceLoadStatistics/website-data-removal-for-site-navigated-to-with-link-decoration-expected.txt: Added.
* LayoutTests/platform/glib/http/tests/resourceLoadStatistics/website-data-removal-for-site-without-user-interaction-expected.txt: Added.
* LayoutTests/platform/mac-monterey/http/tests/resourceLoadStatistics/website-data-removal-for-site-navigated-to-with-link-decoration-expected.txt: Copied from LayoutTests/http/tests/resourceLoadStatistics/website-data-removal-for-site-navigated-to-with-link-decoration-expected.txt.
* LayoutTests/platform/mac-monterey/http/tests/resourceLoadStatistics/website-data-removal-for-site-without-user-interaction-expected.txt: Copied from LayoutTests/http/tests/resourceLoadStatistics/website-data-removal-for-site-without-user-interaction-expected.txt.
* LayoutTests/platform/mac-wk2/TestExpectations:

Canonical link: https://commits.webkit.org/268681@main
